src/devices/Units/Temperature.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Units
+{
+    public struct Temperature
+    {
+        private const double KelvinOffset = 273.15;
+        private const double FahrenheitOffset = 32.0;
+        private const double FahrenheitRatio = 1.8;
+        private double _celsius;
+
+        private Temperature(double celsius)
+        {
+            _celsius = celsius;
+        }
+
+        public double Celsius => _celsius;
+        public double Fahrenheit => FahrenheitRatio * _celsius + FahrenheitOffset;
+        public double Kelvin => _celsius + KelvinOffset;
+
+        public static Temperature FromCelsius(double value)
+        {
+            return new Temperature(value);
+        }
+
+        public static Temperature FromFahrenheit(double value)
```

Comment:
what happen if you pass Nan or Infinity here?

Reply:
It may be interesting to consciously pass NaN through and document/test that it is retained. It could be used for a "no reading" state.